app/models/shipit/deploy_spec/npm_discovery.rb

```diff
@@ -79,7 +92,67 @@ def package_version
         JSON.parse(package_json.read)['version']
       end
 
+      def publish_config
```

Comment:
It feels like we're smooshing a ton of methods into `DeploySpec`.  `PackageJson` and/or `NpmRc` helper classes could have isolated tests, and have less cognitive load for new devs.

I'll defer to Jean because I'm maybe just not embracing what Ruby does well ¯\\_(ツ)_/¯